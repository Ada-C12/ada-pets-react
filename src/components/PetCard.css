  

  
.pet-card {
  margin: 60px 30px 0px 30px;
  display: inline-block;
  min-width: 300px;
    
  .pet-card--body {
    white-space: normal;
  }
}


.pet-card--select-pet-btn {
  margin-left: 15px;
}

.btn {
  margin: 5px;
}


