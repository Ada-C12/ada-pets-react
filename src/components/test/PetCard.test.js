import React from 'react';
import { render, cleanup, getByText } from '@testing-library/react';
import PetCard from '../PetCard';

const petCard = <PetCard
  id={1}
  name={"Samson"}
  species={"Cat"}
  about={"A very awesome cat!  Dont' touch the hair"}
  location={"Seattle, WA"}
  deletePetCallback={() => { }}
  selectPetCallback={() => { }}
/>;

describe('PetCard', () => {
  test('it can be rendered and matches the snapshot', () => {
    const { asFragment } = render(<PetCard
      id={1}
      name={"Samson"}
      species={"Cat"}
      about={"A very awesome cat!  Don't touch the hair"}
      location={"Seattle, WA"}
      deletePetCallback={() => { }}
      selectPetCallback={() => { }}
    />);

    // Assert
    expect(asFragment()).toMatchSnapshot();
    // unmount the rendered content
    cleanup();
  });

  test('It will render the proper name for the pet', () => {
    const container = render(<PetCard
      id={1}
      name={"Samson"}
      species={"Cat"}
      about={"A very awesome cat!  Dont' touch the hair"}
      location={"Seattle, WA"}
      deletePetCallback={() => { }}
      selectPetCallback={() => { }}
    />);

    expect(container.getByText(/Samson/)).toBeDefined();
    expect(container.getByText(/Seattle, WA/).classList).toContain('pet-card--footer');
  });

  test('The "selectPetCallback" function is called when the `select` button is clicked on', () => {

    // Arrange
    // Create a mock callback function
    const selectPet = jest.fn();

    // Render a PetCard
    const container = render(<PetCard
      id={1}
      name={"Samson"}
      species={"Cat"}
      about={"A very awesome cat!  Dont' touch the hair"}
      location={"Seattle, WA"}
      deletePetCallback={() => { }}
      selectPetCallback={selectPet}
    />);

    // Act
    // Find the "Select" button
    const selectButton = container.getByText(/Select/);
    // Trigger a 'click' event
    selectButton.click();

    // Assert
    // Verify that the callback function was called.
    expect(selectPet).toHaveBeenCalled();
  });
});