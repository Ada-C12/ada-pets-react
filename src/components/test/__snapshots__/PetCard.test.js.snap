// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetCard it can be rendered and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="card pet-card"
  >
    <section
      class="pet-card--header"
    >
      <button
        aria-label="Close"
        class="close pet-card--close-btn"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
      ❓ 1 - Samson
      <button
        class="btn btn-primary pet-card--select-pet-btn"
      >
        Select
      </button>
    </section>
    <section
      class="pet-card--body"
    >
      A very awesome cat!  Don't touch the hair
    </section>
    <section
      class="pet-card--footer text-muted"
    >
      Seattle, WA
    </section>
  </div>
</DocumentFragment>
`;
